packages/core/src/model.ts

```diff
@@ -176,7 +176,9 @@ export class ChartModel {
 	setColorScale() {
 		if (this.getDisplayData().datasets[0].fillColors) {
 			this.getDisplayData().datasets.forEach(dataset => {
-				this.colorScale[dataset.label] = scaleOrdinal().range(dataset.fillColors).domain(this.allDataLabels);
+				if (this.allDataLabels) {
+					this.colorScale[dataset.label] = scaleOrdinal().range(dataset.fillColors).domain(this.allDataLabels);
+				}
```

Comment:
Seems like we just need a separate `MeterModel` class that extends `Model` so we don't get the additions here